src/Draco.Compiler/Internal/Symbols/Symbol.cs

```diff
@@ -89,6 +91,46 @@ public virtual string FullName
         }
     }
 
+    /// <summary>
+    /// The fully qualified metadata name of this symbol.
+    /// </summary>
+    public virtual string MetadataFullName
+    {
+        get
+        {
+            var parentFullName = this.ContainingSymbol is not MetadataAssemblySymbol
+                ? this.ContainingSymbol?.MetadataFullName
+                : null;
+
+            return string.IsNullOrWhiteSpace(parentFullName)
+                ? this.MetadataName
+                : $"{parentFullName}.{this.MetadataName}";
+        }
+    }
+
+    /// <summary>
+    /// The full name used to retrieve documentation from metadata.
+    /// </summary>
+    public virtual string DocumentationFullName => this.MetadataFullName;
+
+    /// <summary>
+    /// The documentationPrefix of this type.
+    /// </summary>
+    public string DocumentationPrefix => this switch
+    {
+        TypeSymbol => "T:",
+        ModuleSymbol => "T:",
+        FunctionSymbol => "M:",
+        PropertySymbol => "P:",
+        FieldSymbol => "F:",
+        _ => string.Empty,
+    };
+
+    /// <summary>
+    /// The <see cref="DocumentationFullName"/> with prepended <see cref="DocumentationPrefix"/>.
+    /// </summary>
+    public virtual string PrefixedDocumentationFullName => $"{this.DocumentationPrefix}{this.DocumentationFullName}";
```

Comment:
Since this is reliant on the XML documentation format, I'd move it out of Symbol